cylc/flow/cfgspec/globalcfg.py

```diff
@@ -1442,3 +1452,35 @@ def _no_platform_group_name_overlap(self):
                 for name in names_in_platforms_and_groups:
                     msg += f'\n * {name}'
                 raise GlobalConfigError(msg)
+
+    def platform_dump(
+        self,
+        print_platform_names: bool = True,
+        print_platform_meta: bool = True
+    ) -> None:
+        """Print informations about platforms currently defined.
+        """
+        if print_platform_names:
+            self.dump_platform_names(self)
+        if print_platform_meta:
+            self.dump_platform_details(self)
+
+    @staticmethod
+    def dump_platform_names(cfg) -> None:
+        """Print a list of defined platforms and groups.
+        """
+        platforms = '\n'.join(cfg.get(['platforms']).keys())
+        platform_groups = '\n'.join(cfg.get(['platform groups']).keys())
+        output = (
+            f'{PLATFORM_REGEX_TEXT}\n\n'
```

Comment:
Might be worth sending this text to stderr so users can parse platform names from the CLI.

Reply:
Good idea.